test/test_packages/ArtifactOverrideLoading/Project.toml

```diff
@@ -1,5 +1,5 @@
 name = "ArtifactOverrideLoading"
-uuid = "9f5278ee-142d-5d9a-9723-c4aebaa272e1"
+uuid = "7b879065-7f74-5fa4-bdd5-9b7a15df8941"
```

Comment:
Why did you need to change the UUID here?

Reply:
I changed it to the value of `aol_uuid` because that was the value explicitly given here: https://github.com/JuliaLang/Pkg.jl/blob/f0731405b557b8c04788f7fc6ca03d5a2e673280/test/artifacts.jl#L641. If they are not the same value, Pkg will error out.

Reply:
huh, why wasn't this erroring before?

Reply:
There is a misuse of the API here. API calls were supposed to be made with `Pkg.PackageSpec`, not `Pkg.Types.PackageSpec`. This is needlessly confusing, which is why this PR removes the second constructor.

Reply:
Right, I'm specifically interested in why this having the wrong UUID didn't cause an issue before.

Reply:
Because `Pkg.PackageSpec` did some pre-processing, which was circumvented by using `Pkg.Types.PackageSpec`. In particular, `Pkg.PackageSpec` automatically sets `repo.source`, while `Pkg.Types.PackageSpec` does not. So the package was in some undefined state which just happened to appear like it was tracking a path when it was written to the manifest.

Reply:
So it seems like the intention was that they were supposed to be the same UUID? That is the only thing I was not sure of.

Reply:
Yes, they definitely should have matched

Reply:
Ok, thanks. I think this slight adjustment was all that was necessary